app/code/Magento/Inventory/Plugin/InventoryApi/StockSourceLinksSave/PreventSave/DefaultStockPlugin.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Inventory\Plugin\InventoryApi\StockSourceLinksSave\PreventSave;
+
+use Magento\Framework\App\MaintenanceMode;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\InventoryApi\Api\StockSourceLinksSaveInterface;
+use Magento\InventoryCatalog\Api\DefaultStockProviderInterface;
+
+/**
+ * Prevent saving links related to default stock, except installation process.
+ */
+class DefaultStockPlugin
+{
+    /**
+     * @var MaintenanceMode
+     */
+    private $maintenanceMode;
+
+    /**
+     * @var DefaultStockProviderInterface
+     */
+    private $defaultStockProvider;
+
+    /**
+     * DefaultStockPlugin constructor.
+     *
+     * @param MaintenanceMode $maintenanceMode
+     * @param DefaultStockProviderInterface $defaultStockProvider
+     */
+    public function __construct(
+        MaintenanceMode $maintenanceMode,
+        DefaultStockProviderInterface $defaultStockProvider
+    ) {
+        $this->defaultStockProvider = $defaultStockProvider;
+        $this->maintenanceMode = $maintenanceMode;
+    }
+
+    /**
+     * Prevent saving links related to default stock.
+     *
+     * @param StockSourceLinksSaveInterface $subject
+     * @param \Magento\InventoryApi\Api\Data\StockSourceLinkInterface[] $links
+     * @throws LocalizedException
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function beforeExecute(StockSourceLinksSaveInterface $subject, array $links)
+    {
+        //Exclude installation.
+        if ($this->maintenanceMode->isOn()) {
```

Comment:
We can replace API call in setup script on direct sql (it will be related to creating unchanged initial state)

Reply:
Replace on direct call of resource model, to avoid validation.